client/app/components/Toast/index.jsx

```diff
@@ -0,0 +1,73 @@
+// @flow
+import React, { useState } from 'react';
+import type { Node } from 'react';
+import { I18n } from 'libs/i18n';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTimes } from '@fortawesome/free-solid-svg-icons';
+import css from './Toast.scss';
+
+type Props = {
+  alert?: string,
+  notice?: string,
+  appendDashboardClass?: boolean
+};
+export type State = {
+  showToast: boolean,
+};
+
+export const Toast = ({
+  alert, notice, appendDashboardClass,
+}: Props): Node => {
+  const [showAlert, setShowAlert] = useState<boolean>(alert !== null && alert !== '' && !document.documentElement?.hasAttribute('data-turbolinks-preview'));
+  const [showNotice, setShowNotice] = useState<boolean>(notice !== null && notice !== '' && !document.documentElement?.hasAttribute('data-turbolinks-preview'));
+  const hideNotice = () => {
+    setShowNotice(false);
+  };
+  const hideAlert = () => {
+    setShowAlert(false);
+  };
+  if (showAlert || showNotice) {
+    setTimeout(() => {
+      hideNotice();
+      hideAlert();
+    }, 7000);
+  }
+  return (
+    <>
+      <div id="toast-notice" aria-label={showNotice ? I18n.t('alert_auto_hide') : ''} role="region" aria-live="polite" aria-atomic="true" className={`${showNotice ? 'notice visible' : 'invisible'} ${css.toast} ${showNotice && (showAlert || appendDashboardClass) ? 'smallMarginBottom' : ''}`}>
+        {showNotice && (
+          <>
+            <div>
+              {notice}
+            </div>
+            <button type="button" onClick={hideNotice} aria-label={I18n.t('close')}>
+              <span aria-hidden="true">
+                <FontAwesomeIcon icon={faTimes} />
+              </span>
+            </button>
+          </>
+        )}
+      </div>
+      <div id="toast-alert" aria-label={showAlert ? I18n.t('alert_auto_hide') : ''} role="alert" className={`${showAlert ? 'alert visible' : 'invisible'} ${css.toast} ${showAlert && appendDashboardClass ? 'smallMarginBottom' : ''}`}>
+        {showAlert && (
+          <>
+            <div>
+              {alert}
+            </div>
+            <button type="button" onClick={hideAlert} aria-label={I18n.t('close')}>
+              <span aria-hidden="true">
+                <FontAwesomeIcon icon={faTimes} />
+              </span>
+            </button>
+          </>
+        )}
+      </div>
+    </>
+  );
+};
+
+export default ({
+  alert, notice, appendDashboardClass,
+}: Props): Node => (
+  <Toast alert={alert} notice={notice} appendDashboardClass={appendDashboardClass} />
+);
```

Comment:
The global classes you made for `visible` and `invisible` are not necessary, and you can use conditional JS to hide and show elements!

```suggestion
// @flow
import React, { useState } from 'react';
import type { Node } from 'react';
import { I18n } from 'libs/i18n';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';
import css from './Toast.scss';

type Props = {
  alert?: string,
  notice?: string,
  appendDashboardClass?: boolean,
};
export type State = {
  showToast: boolean,
};

export const Toast = ({ alert, notice, appendDashboardClass }: Props): Node => {
  const [showAlert, setShowAlert] = useState<boolean>(
    alert !== null
      && alert !== ''
      && !document.documentElement?.hasAttribute('data-turbolinks-preview'),
  );
  const [showNotice, setShowNotice] = useState<boolean>(
    notice !== null
      && notice !== ''
      && !document.documentElement?.hasAttribute('data-turbolinks-preview'),
  );
  const hideNotice = () => {
    setShowNotice(false);
  };
  const hideAlert = () => {
    setShowAlert(false);
  };
  if (showAlert || showNotice) {
    setTimeout(() => {
      hideNotice();
      hideAlert();
    }, 7000);
  }
  return (
    <>
      {showNotice && (
        <div
          id="toast-notice"
          aria-label={I18n.t('alert_auto_hide')}
          role="region"
          aria-live="polite"
          aria-atomic="true"
          className={`notice ${css.toast} ${
            showNotice && (showAlert || appendDashboardClass)
              ? 'smallMarginBottom'
              : ''
          }`}
        >
          <>
            <div>{notice}</div>
            <button
              type="button"
              onClick={hideNotice}
              aria-label={I18n.t('close')}
            >
              <span aria-hidden="true">
                <FontAwesomeIcon icon={faTimes} />
              </span>
            </button>
          </>
        </div>
      )}
      {showAlert && (
        <div
          id="toast-alert"
          aria-label={showAlert ? I18n.t('alert_auto_hide') : ''}
          role="alert"
          className={`alert ${css.toast} ${
            appendDashboardClass ? 'smallMarginBottom' : ''
          }`}
        >
          {' '}
          &&
          <>
            <div>{alert}</div>
            <button
              type="button"
              onClick={hideAlert}
              aria-label={I18n.t('close')}
            >
              <span aria-hidden="true">
                <FontAwesomeIcon icon={faTimes} />
              </span>
            </button>
          </>
        </div>
      )}
    </>
  );
};

export default ({ alert, notice, appendDashboardClass }: Props): Node => (
  <Toast
    alert={alert}
    notice={notice}
    appendDashboardClass={appendDashboardClass}
  />
);

```

Reply:
Synced on Slack:

What you have is fine then, can you move the classes for visible and invisible to `client/app/components/Toast/Toast.scss` and rename it to `toastElementVisible` and `toastElementHidden`?

We don't need it in the global stylesheet, since it's not currently being used anywhere else.